tools/serverpod_cli/lib/src/config/config.dart

```diff
@@ -214,6 +231,20 @@ class GeneratorConfig {
       }
     }
 
+    var serializeEnumValuesAsStringsConfig =
+        generatorConfig['serializeEnumValuesAsStrings'];
+    if (serializeEnumValuesAsStringsConfig != null &&
+        serializeEnumValuesAsStringsConfig is! bool) {
+      throw const FormatException(
+          'Option "serializeEnumValuesAsStringsConfig" must be of type bool '
+          'in config/generator.yaml');
+    }
+    var serializeEnumValuesAsStrings =
+        // TODO: switch the default from `false` to `true` in Serverpod 2.0.
+        // Also switch the default in:
+        // tools/serverpod_cli/lib/src/test_util/builders/generator_config_builder.dart
+        (serializeEnumValuesAsStringsConfig as bool?) ?? false;
```

Comment:
Suggestion: add tests for this.

Since this ensure that we don't introduce a breaking change I think we should have a tests that validates that this works as expected.